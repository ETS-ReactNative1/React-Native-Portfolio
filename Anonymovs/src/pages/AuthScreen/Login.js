import React, {useState} from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TextInput,
  View,
  Platform,
  TouchableOpacity,
  KeyboardAvoidingView,
  ActivityIndicator,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import Colors from '../../../assets/config/Colors';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {useDispatch} from 'react-redux';
import {actionLoginName, actionToken} from '../../redux/action/actions';

export default function Login({navigation}) {
  const dispatch = useDispatch();
  const [loginName, setLoginName] = useState('');
  const [loginPush, setLoginPush] = useState(false);

  const loginHandler = () => {
    setLoginPush(true);
    dispatch(actionLoginName(loginName));
    dispatch(actionToken())
    setTimeout(() => {
      navigation.navigate('HomeTabNavigator');
    }, 1000);
  };

  return (
    <KeyboardAvoidingView behavior='padding' style={styles.container}>
      <Image source={require('../../../assets/img/logo-only.png')} />
      <View style={styles.inputWrapper}>
        <TextInput
          //   autoFocus
          placeholder="Input a name to enter"
          placeholderTextColor={Colors.ash}
          style={styles.textInput}
          value={loginName}
          onChangeText={setLoginName}
        />
        <TouchableOpacity
          onPress={() => {
            if (loginName.length <= 3) {
              alert('Input minim 3 chararcter');
            } else {
              loginHandler();
            }
          }}
          style={styles.submitBtn}>
          <Text style={styles.submitTxt}>Submit</Text>
        </TouchableOpacity>
      </View>
      {loginPush == true && (
        <Animatable.View animation="bounceInRight">
          <View style={styles.loginStatus}>
            <Text style={styles.loginName}>You're login as {loginName}</Text>
            <Icon
              name="check-circle-outline"
              size={20}
              color={Colors.success}
            />
          </View>
          <ActivityIndicator size="small" color={Colors.white} />
        </Animatable.View>
      )}
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputWrapper: {
    flexDirection: 'row',
    marginTop: Platform.OS === 'android' ? hp(14) : hp(16),
    alignItems: 'center',
  },
  textInput: {
    borderBottomColor: Colors.ash,
    borderBottomWidth: 1,
    width: wp(55),
    maxWidth: wp(55),
    padding: wp(3),
    color: Colors.white,
  },
  submitBtn: {
    marginLeft: wp(5),
    backgroundColor: Colors.active,
    borderRadius: 4,
    marginTop: Platform.OS === 'android' ? hp(2) : hp(0),
  },
  submitTxt: {
    color: Colors.white,
    padding: wp(3),
  },
  loginName: {
    color: Colors.white,
    marginRight: wp(2),
  },
  loginStatus: {
    flexDirection: 'row',
    marginTop: hp(2),
  },
});
