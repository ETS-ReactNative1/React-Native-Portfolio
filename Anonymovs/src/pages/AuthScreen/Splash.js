import React, {useEffect} from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  StatusBar,
  ActivityIndicator,
} from 'react-native';
import Colors from '../../../assets/config/Colors';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

export default function Splash({navigation}) {
  useEffect(() => {
    setTimeout(() => {
      navigation.navigate('Login');
    }, 1500);
  }, []);

  return (
    <View style={styles.container}>
      <StatusBar barStyle="light-content" />
      <Image source={require('../../../assets/img/logo.png')} />
      <ActivityIndicator
        style={{marginTop: hp(2)}}
        size="large"
        color={Colors.white}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
