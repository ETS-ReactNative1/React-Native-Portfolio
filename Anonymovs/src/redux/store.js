import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import sagas from './saga/sagas';
import combineReducers from './reducer/combine'

const sagaMiddleware = createSagaMiddleware();
const store = createStore(combineReducers,applyMiddleware(sagaMiddleware));

sagaMiddleware.run(sagas);

export default store;