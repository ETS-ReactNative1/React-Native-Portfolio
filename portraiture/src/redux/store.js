import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import combineReducers from './reducers/combiner'
import sagas from './sagas/Sagas'

const sagaMiddleware = createSagaMiddleware();
const store = createStore(combineReducers,applyMiddleware(sagaMiddleware));

sagaMiddleware.run(sagas);

export default store;