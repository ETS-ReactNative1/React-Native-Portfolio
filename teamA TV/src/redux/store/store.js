import ReduxThunk from 'redux-thunk';
import {createStore, applyMiddleware} from 'redux';
import rootReducer from '../reducer/combineReducer'


export let store = createStore(rootReducer,applyMiddleware(ReduxThunk))
