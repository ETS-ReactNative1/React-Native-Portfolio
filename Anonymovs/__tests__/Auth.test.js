import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../src/redux/store';

import Splash from '../src/pages/AuthScreen/Splash';
import Login from '../src/pages/AuthScreen/Login';

jest.useFakeTimers();

test('Profile SnapShoot', () => {
  const snap = renderer
    .create(
      <Provider store={store}>
        <Splash />
      </Provider>,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
test('AboutApp SnapShoot', () => {
  const snap = renderer
    .create(
      <Provider store={store}>
        <Login />
      </Provider>,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Splash />
      </Provider>,
    )
    .toJSON();
  expect(tree.children.length).toBe(2);
});
it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Login />
      </Provider>,
    )
    .toJSON();
  expect(tree.children.length).toBe(2);
});
