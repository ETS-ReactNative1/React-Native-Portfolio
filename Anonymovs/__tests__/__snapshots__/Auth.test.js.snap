// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutApp SnapShoot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#191926",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/img/logo-only.png",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 213.5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Input a name to enter"
      placeholderTextColor="#999999"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomColor": "#999999",
          "borderBottomWidth": 1,
          "color": "#FFFFFF",
          "maxWidth": 412.5,
          "padding": 22.5,
          "width": 412.5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FF3365",
          "borderRadius": 4,
          "marginLeft": 37.5,
          "marginTop": 0,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "padding": 22.5,
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`Profile SnapShoot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#191926",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/img/logo.png",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#FFFFFF"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "marginTop": 26.5,
      }
    }
  />
</View>
`;
